apis/management.cattle.io/v3/logging_types.go

```diff
@@ -87,47 +87,35 @@ type LoggingCondition struct {
 }
 
 type ElasticsearchConfig struct {
-	Host         string `json:"host,omitempty"`
-	Port         int    `json:"port,omitempty"`
-	IndexPrefix  string `json:"indexPrefix,omitempty"`
-	DateFormat   string `json:"dateFormat,omitempty"`
+	Endpoint     string `json:"endpoint,omitempty" norman:"required"`
+	IndexPrefix  string `json:"indexPrefix,omitempty" norman:"required"`
+	DateFormat   string `json:"dateFormat,omitempty" norman:"required,type=enum,options=YYYY-MM-DD|YYYY-MM|YYYY,default=YYYY-MM-DD"`
 	AuthUserName string `json:"authUsername,omitempty"` //secret
 	AuthPassword string `json:"authPassword,omitempty"` //secret
 }
 
 type SplunkConfig struct {
-	Host     string `json:"host,omitempty"`
-	Port     int    `json:"port,omitempty"`
-	Protocol string `json:"protocol,omitempty"`
+	Endpoint string `json:"endpoint,omitempty" norman:"required"`
 	Source   string `json:"source,omitempty"`
-	Token    string `json:"token,omitempty"` //secret
+	Token    string `json:"token,omitempty" norman:"required"` //secret
 }
 
 type EmbeddedConfig struct {
-	IndexPrefix string `json:"indexPrefix,omitempty"`
-	DateFormat  string `json:"dateFormat,omitempty"`
+	IndexPrefix           string `json:"indexPrefix,omitempty" norman:"required"`
+	DateFormat            string `json:"dateFormat,omitempty" norman:"required,type=enum,options=YYYY-MM-DD|YYYY-MM|YYYY,default=YYYY-MM-DD"`
+	ElasticsearchEndpoint string `json:"elasticsearchEndpoint,omitempty" norman:"nocreate"`
+	KibanaEndpoint        string `json:"kibanaEndpoint,omitempty" norman:"nocreate"`
 }
 
 type KafkaConfig struct {
-	Zookeeper      *Zookeeper  `json:"zookeeper,omitempty"`
-	Broker         *BrokerList `json:"broker,omitempty"`
-	Topic          string      `json:"topic,omitempty"`
-	DataType       string      `json:"dataType,omitempty"`
-	MaxSendRetries int         `json:"maxSendRetries,omitempty"`
-}
-
-type Zookeeper struct {
-	Host string `json:"host,omitempty"`
-	Port int    `json:"port,omitempty"`
-}
-
-type BrokerList struct {
-	BrokerList []string `json:"brokerList,omitempty"`
+	ZookeeperEndpoint string `json:"zookeeperEndpoint,omitempty"`
+	BrokerEndpoints   string `json:"brokerEndpoints,omitempty"`
```

Comment:
there can be multiple endpoints for brokerEndpoints? if so, any reason why we don't use []string type?

Reply:
it can be multiple endpoints for brokerEndpoints, I can change to use []string.